pyface/ui/wx/tasks/editor_area_pane.py

```diff
@@ -137,14 +137,14 @@ def _get_editor_with_control(self, control):
 
     # Trait change handlers ------------------------------------------------
 
-    @on_trait_change("editors:[dirty, name]")
-    def _update_label(self, editor, name, new):
-        index = self.control.GetPageIndex(editor.control)
-        self.control.SetPageText(index, self._get_label(editor))
+    @observe("editors:items:[dirty, name]")
+    def _update_label(self, event):
+        index = self.control.GetPageIndex(event.object.control)
+        self.control.SetPageText(index, self._get_label(event.object))
 
-    @on_trait_change("editors:tooltip")
-    def _update_tooltip(self, editor, name, new):
-        self.control.SetPageToolTip(editor.control, editor.tooltip)
+    @observe("editors:items:tooltip")
+    def _update_tooltip(self, event):
+        self.control.SetPageToolTip(event.objecet.control, event.object.tooltip)
```

Comment:
similar comments as before - replacing `event.object` with something meaningful like `editor` in this case - i think. i'm not sure if `event.object` is `editor` though.